### ISSUE TITLE
nvda threshold do not advertise the typed words when pricing space

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#9577]
### Steps to reproduce:
in the nvda keyboard settings, check the 2 options:
speak typed characters
speak typed words
Open Notepad and type a word, press spacebar
### Actual behavior:
nvda announces only space and does not pronounce the word that has just been typed.
### Expected behavior:
nvda should announce the typed space character and then announce the typed word.
#### NVDA installed/portable/running from source:
install
#### NVDA version:
threshold 17249
#### Windows version:
10 17763.475
#### Other information about your system:
If you disable
speak typed characters
By pressing space, nvda announces the word that has just been typed.
nvda is speaking the space character and is suppressing the reading of the typed word.
using Portuguese one-core voice Brazil, voice Maria.

#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
yes
nvda alfa does not have this problem

**Comment by leonardder** [comment]
I can reproduce this issue.

Cc @michaelDCurran 

Marking this as p2, but a p1 might be more sufficient here, since this regards essential functionality many people are using.

**Comment by michaelDCurran** [comment]
speech.speak always clears curWordChars (forgetting the current word being typed). However, speech.speakSpelling (used for speaking a typed character) internally calls speech.speak. I think before speech refactor speakSpelling bypassed speech.speak and called the synth's speak method directly.
We originally chose to clear curWordChars in speech.speak to lessen the chance that typing a non-alphaNumeric character after other actions would not cause a word from a long time ago to suddenly speak.
  The most important thing here is to no longer clear curWordChars in speech.speak. But, we'll need to ensure that curWordChars *is* cleared on all non-typing related actions. E.g. focus changes, and caret navigation.

**Comment by michaelDCurran** [comment]
Of course there is pr #8110 that replaces all of this code anyway. But right now at least, we should probably fix this current implementation.

**Comment by leonardder** [comment]
I believe #8110 still relies on curWordChars not being cleared, though I"m not exactly sure.

I'd appreciate it if you could keep #7812 in mind when looking in to this. That was actually the trigger to create #8110 in the first place.
